src/aap_eda/tasks/project.py

```diff
@@ -56,15 +83,38 @@ def _import_project(project_id: int):
     logger.info(f"Task complete: Import project ( project_id={project.id} )")
 
 
-@job(PROJECT_TASKS_QUEUE)
-def sync_project(project_id: int):
-    with advisory_lock(f"import_project_{project_id}", wait=False) as acquired:
+def sync_project(project_id: int) -> tp.Optional[str]:
+    """Sync project async task.
+
+    Proxy for sync_project_dispatcherd and sync_project_rq.
+    """
+    with advisory_lock(f"sync_project_{project_id}", wait=False) as acquired:
         if not acquired:
             logger.debug(
                 f"Another task already syncing project {project_id}, exiting"
             )
-            return
-        _sync_project(project_id)
+            return None
+        if features.DISPATCHERD:
+            return sync_project_dispatcherd(project_id)
+        # rq
+        job_data = sync_project_rq(project_id)
+        return job_data.id
+
+
+def sync_project_dispatcherd(project_id: int) -> str:
+    queue_name = utils.sanitize_postgres_identifier(PROJECT_TASKS_QUEUE)
+    job_data, queue = submit_task(
+        _sync_project,
+        queue=queue_name,
+        args=(project_id,),
+    )
+    return job_data["uuid"]
+
+
+@tasking.redis_connect_retry()
+def sync_project_rq(project_id: int):
+    queue = django_rq.get_queue(name=PROJECT_TASKS_QUEUE)
+    return queue.enqueue(_sync_project, project_id=project_id)
```

Comment:
@Alex-Izquierdo can this return id to be consistent with sync_project_dispatcherd
```
     return queue.enqueue(_sync_project, project_id=project_id).id
```

Reply:
Good point, thanks. Addressed in https://github.com/ansible/eda-server/pull/1320, please review.